control-plane/agents/src/bin/core/tests/volume/format_options.rs

```diff
@@ -0,0 +1,157 @@
+#![cfg(test)]
+
+use deployer_cluster::ClusterBuilder;
+use stor_port::types::v0::openapi::models;
+use stor_port::types::v0::transport::VolumeId;
+
+use std::{collections::HashMap, process::Command, time::Duration};
+
+const VOLUME_SIZE: u64 = 32u64 * 1024 * 1024;
+const POOL_SIZE: u64 = 140u64 * 1024 * 1024;
+const BYTES_PER_INODE: u64 = 64u64 * 1024;
+
+struct DeviceDisconnect(nvmeadm::NvmeTarget);
+impl Drop for DeviceDisconnect {
+    fn drop(&mut self) {
+        if self.0.disconnect().is_err() {
+            std::process::Command::new(env!("SUDO"))
+                .args(["nvme", "disconnect-all"])
+                .status()
+                .unwrap();
+        }
+    }
+}
+
+/// This test.
+/// Create a volume and format using formatOption = bytes-per-inode(determines no. of inodes).
+/// Mounts the fs onto the mount path inside csi-node container.
+/// Inspects tune2fs output for the device and ensure inode count is correct.
+#[tokio::test]
+async fn filesystem_volume_format_options() {
+    let cache_period = Duration::from_millis(250);
+    let reconcile_period = Duration::from_millis(3000);
+    let cluster = ClusterBuilder::builder()
+        .with_rest(true)
+        .with_io_engines(1)
+        .with_tmpfs_pool_ix(0, POOL_SIZE)
+        .with_csi(true, true)
+        .with_cache_period(&humantime::Duration::from(cache_period).to_string())
+        .with_reconcile_period(reconcile_period, reconcile_period)
+        .build()
+        .await
+        .expect("Failed to build cluster");
+
+    let api_client = cluster.rest_v00();
+    let volumes_api = api_client.volumes_api();
+    let vol_id = VolumeId::new();
+    let volume = volumes_api
+        .put_volume(
+            &vol_id,
+            models::CreateVolumeBody::new(
+                models::VolumePolicy::new(true),
+                1,
+                VOLUME_SIZE,
+                false,
+                false,
+            ),
+        )
+        .await
+        .expect("Failed to create volume");
+
+    let mut node = cluster
+        .csi_node_client(0)
+        .await
+        .expect("Failed to get node client");
```

Comment:
```suggestion
        .expect("To get node client");
```
same throughout, though a simple unwrap also works for this test code